### ISSUE TITLE
Time Series Transformer doesn't converge

### ISSUE THREAD

**Issue by Eran-BA** [KasperGroesLudvigsen/influenza_transformer#13]
I have a question, I was able to run the training loop but the transformer doesn't converge,
is it because of the dataset? or it is because of some wrong hyper parameters or a chosen loss function.

thanks.

**Comment by Aimen1996** [comment]
same question

**Comment by Athuliva** [comment]
the gradient is accumulating, clear the gradient of the previous step using `optimizer.zero_grad()` 
also I am getting a flat line as output if I infer the model in `model.eval()` mode . It seems there is some sort of bug in transformers encoder layer, source : https://discuss.pytorch.org/t/model-eval-predicts-all-the-same-or-nearly-same-outputs/155025/11

**Comment by rFalque** [comment]
Same here, that would be interesting to provide an example of a dataset where the forecasting works (I can predict anything aside from the mean value).
